plugins/aws/resoto_plugin_aws/resource/redshift.py

```diff
@@ -0,0 +1,445 @@
+from typing import ClassVar, Dict, Optional, List
+
+from attrs import define, field
+
+from resoto_plugin_aws.resource.base import AwsResource, AwsApiSpec
+from resotolib.json_bender import Bender, S, Bend, ForallBend, K, F
+from resoto_plugin_aws.aws_client import AwsClient
+from resoto_plugin_aws.utils import ToDict
+from typing import Tuple, Type
+from datetime import datetime
+
+
+@define(eq=False, slots=False)
+class AwsRedshiftNetworkInterface:
+    kind: ClassVar[str] = "aws_redshift_network_interface"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "network_interface_id": S("NetworkInterfaceId"),
+        "subnet_id": S("SubnetId"),
+        "private_ip_address": S("PrivateIpAddress"),
+        "availability_zone": S("AvailabilityZone"),
+    }
+    network_interface_id: Optional[str] = field(default=None)
+    subnet_id: Optional[str] = field(default=None)
+    private_ip_address: Optional[str] = field(default=None)
+    availability_zone: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsRedshiftVpcEndpoint:
+    kind: ClassVar[str] = "aws_redshift_vpc_endpoint"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "vpc_endpoint_id": S("VpcEndpointId"),
+        "vpc_id": S("VpcId"),
+        "network_interfaces": S("NetworkInterfaces", default=[]) >> ForallBend(AwsRedshiftNetworkInterface.mapping),
+    }
+    vpc_endpoint_id: Optional[str] = field(default=None)
+    vpc_id: Optional[str] = field(default=None)
+    network_interfaces: List[AwsRedshiftNetworkInterface] = field(factory=list)
+
+
+@define(eq=False, slots=False)
+class AwsRedshiftEndpoint:
+    kind: ClassVar[str] = "aws_redshift_endpoint"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "address": S("Address"),
+        "port": S("Port"),
+        "vpc_endpoints": S("VpcEndpoints", default=[]) >> ForallBend(AwsRedshiftVpcEndpoint.mapping),
+    }
+    address: Optional[str] = field(default=None)
+    port: Optional[int] = field(default=None)
+    vpc_endpoints: List[AwsRedshiftVpcEndpoint] = field(factory=list)
+
+
+@define(eq=False, slots=False)
+class AwsRedshiftClusterSecurityGroupMembership:
+    kind: ClassVar[str] = "aws_redshift_cluster_security_group_membership"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "cluster_security_group_name": S("ClusterSecurityGroupName"),
+        "status": S("Status"),
+    }
+    cluster_security_group_name: Optional[str] = field(default=None)
+    status: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsRedshiftVpcSecurityGroupMembership:
+    kind: ClassVar[str] = "aws_redshift_vpc_security_group_membership"
+    mapping: ClassVar[Dict[str, Bender]] = {"vpc_security_group_id": S("VpcSecurityGroupId"), "status": S("Status")}
+    vpc_security_group_id: Optional[str] = field(default=None)
```

Comment:
should this result in a `dependant_node` to the corresponding instance of `AwsEc2SecurityGroup`?